### ISSUE TITLE
[Question] what are makeCons and makeElmishCons?

### ISSUE THREAD

**Issue by OnurGumus** [fable-compiler/Fable.Store#14]
Hi, I peeked the source code but still I am a bit confused. Could you explain what are makeCons and makElmishCons?

**Comment by alfonsogarciacaro** [comment]
Yes, sorry, this should be either private or have a better explanation. It took also me a while to remember why I wrote those functions 😅

The cons(tructor) is just a function that returns a store. The idea is that Store could be a generic value. Why? Because at the beginning Fable.Store was written for interop with Svelte and Svelte has its own stores. This way Fable Svelte apps can [write Elmish apps](https://github.com/fable-compiler/Fable.Store/blob/4ebeff41786147deb93ea427e2dd4a6e2d6f958b/samples/App/src/TodoMVC/TodoMVC.fs#L53-L56) that could be [sent to Svelte](https://github.com/fable-compiler/Fable.Store/blob/4ebeff41786147deb93ea427e2dd4a6e2d6f958b/samples/App/src/TodoMVC/TodoMVC.svelte#L8) as a normal Svelte store.

However, when we wanted to extend Fable.Store to other apps we wanted to make `IStore` compatible with `IObservable` instead so we could take advantage of Observable pipelines (though unfortunately I haven't seen many examples of this) so we implemented our own store interface. This is way when you call `Store.makeElmish` an `IStore` is created by default, but if you use `SvelteStore.makeElmish` an [Svelte store](https://github.com/fable-compiler/Fable.Store/blob/4ebeff41786147deb93ea427e2dd4a6e2d6f958b/src/Fable.SvelteStore/Fable.SvelteStore.fs#L29-L35) is used instead.
